.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+name: Tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
```

Comment:
I might be mistaken because I am not sure what is the best here, but wouldn't it be useful to also trigger it on push to other branches? That way we ensure that our code is good before opening the pr 🤔